### ISSUE TITLE
bad allocation

### ISSUE THREAD

**Issue by jpstotz** [azerg/NppPlistPlugin#6]
When opening a large file in Notepad++ (64bit 7.8.5) while NppBplistPlugin v1.2.0.4 is installed multiple error dialogs pup-up when the file is loaded and if 

Steps to reproduce:

1. Unzip the attached file `zero.bin`
2. Make sure Notepad++ is not running
3. Select "Edit with Notepad++" in the file's context menu
4. The error popups are shown:

Title: Notepad++ bplist bad allocation
Text: bad allocation

So there are two questions:

1. Why is the plugin active on non-plist file?
2. What memory does it try to allocate?

The following file triggers this bug (unzip and open with Notepad++ with NppBplistPlugin installed). It is a 360MB file with only zero data.
[zero.zip](https://github.com/azerg/NppBplistPlugin/files/4676355/zero.zip)

I know this file does not make much sense however however a user may open such a file accidentally. In any way a plugin should not affect Notepad++ in a negative way independent what file the user loads.

**Comment by azerg** [comment]
oh. Thanks for the sample file!

**Comment by azerg** [comment]
Validation is pretty simple
```cpp
 template <typename T>
  bool IsValidBplist( T pFileBuff )
  {
    return ( pFileBuff.size() > sizeof( defs::g_szHeader ) ) &&
      std::equal( std::begin( defs::g_szHeader ), std::end( defs::g_szHeader ), pFileBuff.begin() );
  }
```

and goes before almost all stuff.
Reproduced the bug - saw the same bad_alloc from entire notepad++, before the same one from Bplist plugin. Will  fix it today, also it takes some time to validate plugin update by owner.

**Comment by azerg** [comment]
@jpstotz , fixed and published release with binaries. Pushing update to notepad++ will take a bit, but you could download and replace manually, or wait for integration with npp and their plugin manager.

**Comment by jpstotz** [comment]
@azerg Thanks for fixing this issue. We will see when the fixed version v1.3.0.0 will be available via Notepad++ PluginManager.
